src/fromager/overrides.py

```diff
@@ -82,7 +89,14 @@ def extra_environ_for_pkg(
         with open(env_file, "r") as f:
             for line in f:
                 key, _, value = line.strip().partition("=")
-                extra_environ[key.strip()] = value.strip()
+                key = key.strip()
+                value = value.strip()
+                if "$(" in value:
```

Comment:
Should we worry about the case where `"$("` might be escaped out and just part of the value as a string literal?

Reply:
That's a good question.

I wonder how often that is likely to come up in practice? How hard would it be to detect that case?

Reply:
Let's take this as it is, and consider the improvement as another PR. I'll open a ticket for it.